app/models/product_import/entry_processor.rb

```diff
@@ -0,0 +1,234 @@
+module ProductImport
+  class EntryProcessor
+    attr_reader :inventory_created, :inventory_updated, :products_created, :variants_created, :variants_updated, :products_reset_count, :supplier_products, :total_supplier_products
+
+    def initialize(importer, validator, import_settings, spreadsheet_data, editable_enterprises, import_time, updated_ids)
+      @importer = importer
+      @validator = validator
+      @import_settings = import_settings
+      @spreadsheet_data = spreadsheet_data
+      @editable_enterprises = editable_enterprises
+      @import_time = import_time
+      @updated_ids = updated_ids
+
+      @inventory_created = 0
+      @inventory_updated = 0
+      @products_created = 0
+      @variants_created = 0
+      @variants_updated = 0
+      @products_reset_count = 0
+      @supplier_products = {}
+      @total_supplier_products = 0
+    end
+
+    def save_all(entries)
+      entries.each do |entry|
+        if import_into_inventory?(entry)
+          save_to_inventory(entry)
+        else
+          save_to_product_list(entry)
+        end
+      end
+
+      @importer.errors.add(:importer, I18n.t(:product_importer_products_save_error)) if total_saved_count.zero?
+    end
+
+    def count_existing_items
+      @spreadsheet_data.suppliers_index.each do |_supplier_name, supplier_id|
+        next unless supplier_id && permission_by_id?(supplier_id)
+
+        products_count =
+          if import_into_inventory_by_supplier?(supplier_id)
+            VariantOverride.where('variant_overrides.hub_id IN (?)', supplier_id).count
+          else
+            Spree::Variant.
+              joins(:product).
+              where('spree_products.supplier_id IN (?)
+              AND spree_variants.is_master = false
```

Comment:
you could extract this into a scope such as `non_master` and reuse it in `#product_import_dates`. This is the added benefit of using AR vs plain strings.